shunting-yard-common/src/test/java/com/hotels/shunting/yard/common/io/jackson/DeserializerWithJsonInputTest.java

```diff
@@ -0,0 +1,156 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.shunting.yard.common.io.jackson;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.util.List;
+import java.util.Map;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import com.hotels.shunting.yard.common.event.AddPartitionEvent;
+import com.hotels.shunting.yard.common.event.AlterPartitionEvent;
+import com.hotels.shunting.yard.common.event.CreateTableEvent;
+import com.hotels.shunting.yard.common.event.DropPartitionEvent;
+import com.hotels.shunting.yard.common.event.DropTableEvent;
+import com.hotels.shunting.yard.common.event.EventType;
+import com.hotels.shunting.yard.common.event.InsertTableEvent;
+import com.hotels.shunting.yard.common.event.ListenerEvent;
+
+@RunWith(MockitoJUnitRunner.class)
+public class DeserializerWithJsonInputTest {
+  private final JsonMetaStoreEventDeserializer metaStoreEventDeserializer = new JsonMetaStoreEventDeserializer();
+  private final ApiarySqsMessageDeserializer sqsDeserializer = new ApiarySqsMessageDeserializer(
+      metaStoreEventDeserializer);
+
+  private final static String BASE_EVENT_FROM_SNS = "{"
+      + "  \"Type\" : \"Notification\","
+      + "  \"MessageId\" : \"9b0f34bc-ae00-57c4-97a0-60f5b002b3d0\","
+      + "  \"TopicArn\" : \"arn:aws:sns:us-west-2:440407435941:abhimanyu-sns-test\","
+      + "  \"Timestamp\" : \"2018-10-23T13:01:54.507Z\","
+      + "  \"SignatureVersion\" : \"1\","
+      + "  \"Signature\" : \"P9vAm5YsTRZkQOcgJ5YEkyTAwppGE8G8Y018RMUMLFiRpsSTJ+DGErNiMwz+qUv4RyBg3yEnwK0Nc+OTAcgkc9RARLN0OpoWG7cYt+N/m6orrDqe33EA7krocYiO+a6+lVu3/oNVUZVvBZ+mahizSRRnCzVaJszFhvpPS3rYCfssI/1zpx9s6gMpUehpU7ZK7DCTvG7zsfYIUO1Df/mMdESrO19pVOfYjVavk6K3nU+y+1TmDduEcBaUFzOLmTRAxprgmoq1JrTJMl0V5TOkMdWFn3+hzx/5q82RDQdmAVVKZWy/nfiuwB/S5YZOJywP6EaudaX1wgwpdhE0RTqVXQ==\","
+      + "  \"SigningCertURL\" : \"https://sns.us-west-2.amazonaws.com/SimpleNotificationService-ac565b8b1a6c5d002d285f9598aa1d9b.pem\","
+      + "  \"UnsubscribeURL\" : \"https://sns.us-west-2.amazonaws.com/?Action=Unsubscribe&SubscriptionArn=arn:aws:sns:us-west-2:440407435941:abhimanyu-sns-test:fc4bd684-c38b-4eaf-a718-fac879c6a996\",";
+
+  private final static String ADD_PARTITION_EVENT = "\"Message\" : \"{\\\"protocolVersion\\\":\\\"1.0\\\",\\\"eventType\\\":\\\"ADD_PARTITION\\\",\\\"dbName\\\":\\\"some_db\\\",\\\"tableName\\\":\\\"some_table\\\",\\\"partitionKeys\\\":{\\\"col_1\\\": \\\"string\\\", \\\"col_2\\\": \\\"integer\\\",\\\"col_3\\\": \\\"string\\\"},\\\"partitionValues\\\":[\\\"val_1\\\", \\\"val_2\\\", \\\"val_3\\\"]}\"";
```

Comment:
I saw some parameterized tests in the code which were serializing a java representation but I found writing these tests extremely important in understanding how Shunting Yard works as there is no way anyone will be able to see what JSON SY receives without these tests. I am sure any new person with no prior knowledge of SY will appreciate the presence of these tests here.

I agree with you about reading it from the file but because its such a simple test and the only one in SY where I am actually firing a JSON to build an event, I am going to leave it as is. If I have one more place where I need to do something like this, I will move these to file.